spyder/widgets/variableexplorer/dataframeeditor.py

```diff
@@ -399,23 +401,102 @@ def reset(self):
+            setResizeMode(QHeaderView.Fixed)
+
+        parent.viewport().stackUnder(self)
+
+        self.setSelectionModel(parent.selectionModel())
+        for col in range(1, parent.model().columnCount()):
+            self.setColumnHidden(col, True)
+
+        self.setColumnWidth(0, parent.columnWidth(0))
+        self.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.show()
+
+        self.setVerticalScrollMode(1)
+
+
 class DataFrameView(QTableView):
     """Data Frame view class"""
     def __init__(self, parent, model):
         QTableView.__init__(self, parent)
         self.setModel(model)
 
+        self.frozenTableView = FrozenTableView(self)
+        self.updateFrozenTableGeometry()
+
+        self.setHorizontalScrollMode(1)
+        self.setVerticalScrollMode(1)
+
+        self.horizontalHeader().sectionResized.connect(self.updateSectionWidth)
+        self.verticalHeader().sectionResized.connect(self.updateSectionHeight)
+
+        self.frozenTableView.verticalScrollBar().valueChanged.connect(
+            self.verticalScrollBar().setValue)
+
         self.sort_old = [None]
         self.header_class = self.horizontalHeader()
         self.header_class.sectionClicked.connect(self.sortByColumn)
         self.menu = self.setup_menu()
         config_shortcut(self.copy, context='variable_explorer', name='copy',
                         parent=self)
         self.horizontalScrollBar().valueChanged.connect(
-                            lambda val: self.load_more_data(val, columns=True))
+                        lambda val: self.load_more_data(val, columns=True))
+        self.verticalScrollBar().valueChanged.connect(
+                        lambda val: self.load_more_data(val, rows=True))
         self.verticalScrollBar().valueChanged.connect(
-                               lambda val: self.load_more_data(val, rows=True))
+            self.frozenTableView.verticalScrollBar().setValue)
     
+    def updateSectionWidth(self, logicalIndex, oldSize, newSize):
+        if logicalIndex == 0:
+            self.frozenTableView.setColumnWidth(0, newSize)
+            self.updateFrozenTableGeometry()
+
+    def updateSectionHeight(logicalIndex, oldSize, newSize):
+        self.frozenTableView.setRowHeight(logicalIndex, newSize)
+
+    def resizeEvent(self, event):
+        QTableView.resizeEvent(self, event)
+        self.updateFrozenTableGeometry()
+
+    def moveCursor(self, cursorAction, modifiers):
+        current = QTableView.moveCursor(self, cursorAction, modifiers)
+        if cursorAction == self.MoveLeft and \
+                           current.column() > 1 and \
+                           self.visualRect(current).topLeft().x() < \
+                           (self.frozenTableView.columnWidth(0) + 
+                            self.frozenTableView.columnWidth(1)):
+            newValue = self.horizontalScrollBar().value() + \

```

Comment:
pep8 for this variable, and there's no need of continuation lines if you enclose this whole expression in parenthesis